modules/aws-eks-teams/main.tf

```diff
@@ -50,10 +52,25 @@ resource "aws_iam_role" "team_access" {
   tags = var.tags
 }
 
-resource "kubernetes_cluster_role" "team" {
+# ---------------------------------------------------------------------------------------------------------------------
+# Application Team EKS Access IAM Policy
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_iam_policy" "application_team" {
+  count = length(var.application_teams) > 0 ? 1 : 0
+
+  name        = "${var.eks_cluster_id}-ApplicationTeamEKSAccess"
+  path        = "/"
+  description = "Application Team EKS Access IAM Policy"
+  policy      = var.application_team_iam_policy
+
+  tags = var.tags
+}
+
```

Comment:
Can you please share more about this use case?
I assume you wanted to provide EKS Access (UI) to your application team, but do you also want to provide custom policy for your use case?
And if so, this means that every team using the same module will have this policy applied to the role. the name of the role and the policy provided is around "Access", but access to what?
So, the way I think about this is:
1. we can have similar "access" to EKS UI as the platform team (need to confirm that the team will only see resources relevant to the team via the UI, should be since it's provided via the k8s auth map setup) with default policy or allow users to provide custom one.
Also note the name of the policy which is "ApplicationTeamEKSAccess".
2. on line 37 here, we apply the same provided policy to all the teams, this might not be what customer expect to happen.

Reply:
1. Actually this is not related with console UI access for team(person), but I followed iam_policy name for platform team.

I need to grant some permissions to application, so it's serviceaccount, for example,
access s3, update cloudfront distribution, create route53 hostedzone, etc.
This web application uses/controls AWS resources.

2. In my original issue, this is related with Q1. In my case, one application team use two namespaces,
so this is okay for now.

I think that your team need to decide whether to allow multiple application teams to exist,
and they can use one or multiple namespaces, and how to implement this. That will be a fundamental solution.

Reply:
Maybe I misunderstood something but If your use case is for web application, why this is related to Teams?
Teams by design is for human access and interaction with the cluster, allows a team to access their "environment" (AKA namespace/s).

I disagree with point number 2,  we should think big here for overall usage, going with this direction limits any other customers who want to use this module where a single policy will always apply to all other application teams, this is very opinionated approach that we want to be careful with. 
We should have a better design to approach your ask on point 2 IMO, this current approach could be a feature flag to apply same policy for all application teams.